### DISCUSSION TITLE
Feedback on the last version of the ADF

### DISCUSSION THREAD

**Discussion by cecilehannay** [NCAR/ADF#189] [General]
Please have a look at the latest version of the [ADF diags](https://webext.cgd.ucar.edu/FWscHIST/f.cesm3_cam058_mom_b.FWscHIST.ne30_L58.001/atm/ADF/f.cesm3_cam058_mom_b.FWscHIST.ne30_L58.001_1980_1989_vs_f.cesm3_cam058_mom.FWscHIST.ne30_L58.001_1980_1989/)

Please provide feedback on:
- web design
- categories ([spreadsheet](https://docs.google.com/spreadsheets/d/17IxS8Du8wkLWefzELex3e3RCr3GrpSr6o5D3GZm8xto/edit?usp=sharing) to edit categories)
- colorbars
- etc... 

---- 

Request feedback from: @JulioTBacmeister, @andrewgettelman, @swrneale, @adamrher, ...

**Comment by cecilehannay** [comment]
Justin's slides presented at the CAM7 meeting: [2022-06-24 ADF Update.pdf](https://github.com/NCAR/ADF/files/8977416/2022-06-24.ADF.Update.pdf)

**Comment by swrneale** [comment]
Here's my webpage observations/suggestions, based mainly on the 'what is stopping me moving from the old diags.' point of view.  Some are small things I know.


-For the webpage in general I think we should try and fit as much of the plot and the links onto a standard laptop screen resolution as possible. I'm sure we could come up with clever ways to avoid scrolling or having to resize the image.
-For the variable long names, I think it might be useful to take advantage of 'on mouseover' to map on top of the short name as newer users may not know the short name well (and it is written explicitly in the old diags.).
-However scrolling can be avoided on the variable lists would be awesome.  I think for the left panel, having links right at the top for sub groups (Clouds/Budget etc.) would reduce scrolling. Heck you may want to sub-group for each season so that JJA/DJF/.. is linked for every single variable. 
-I think we may want to distinguish the font/size of the text differently for the titles/sim. names/plot types. I also think the mouse-over text would be useful for a description of each plot type.
-The way that the inset plots stay stationary is a bit weird. My natural expectation is that when I click on the plot link to the left I can scroll the plot up or down if it gets cutoff. And if I click on another plot link on the left panel the list of plots should remain stationary, but it resets to the top. This can be a bit cumbersome to keep scrolled back down to where you were.
-For the tables. Justin may have done this, but I think a constructed png is better for this as we can expend it's functionality, in terms of highlighting, later.
-For the landing page I do like images that are there for all the sets to be clicked, I did use them.

Thanks for everyone's effort on this.
Rich

**Reply by justin-richling** [reply]
Thanks for the input @swrneale, I appreciate you taking time to give us your thoughts and I agree the current layout is far from ideal. Now that we have a working website, we can start fine tuning it.

**Comment by swrneale** [comment]
@justin-richling  No problem. Of course they're just comments and we can chat if not clear, which now I read them again some are not that clear  :) !

**Reply by dan800** [reply]
An option or subset for log-pressure coordinates for the zonal plots would be beneficial for a number of fields, e.g., CH4, GW tendencies, stratospheric temperature.

**Reply by brianpm** [reply]
@dan800 -- I've got a PR ready to go that allows any variable to be plotted in log-pressure for both zonal and meridional contour plots (https://github.com/NCAR/ADF/pull/188). The way it works is very simple: in the variable defaults YAML file, if a variable has an entry: `plot_log_pressure: True` then it will get plotted on log-p using Matplotlib's `set_yscale("log")` method. 

More customized plots can be made as add-on scripts (e.g., https://github.com/brianpm/ADF/blob/marshian_plot/scripts/plotting/explore_marshian.ipynb).

**Reply by dan800** [reply]
Cool. Thanks so much for working on this. Looks like a nice way to set the scale. I had a quick look at your PR and saw this:

 if cp_info['plot_log_p']:
             print("SWITCH TO LOG PRESSURE!")
             [a.set_yscale("log") for a in ax]
             fig.text(-0.03, 0.5, 'LOG(PRESSURE [hPa])', va='center', rotation='vertical')
         else:
             print(f"THE OPTION SET: {cp_info['plot_log_p']}")
             fig.text(-0.03, 0.5, 'PRESSURE [hPa]', va='center', rotation='vertical')

I don't think you need to change the axis label - it's still Pressure [hPa], just spaced differently (unless the tick label values really are 3, 2, 1, 0, -1, etc. for 1000, 100, 10, 1, 0.1)

**Comment by justin-richling** [comment]
I have made a quick prototype for a different website layout, please take a look [here](https://project.cgd.ucar.edu/projects/ADF/webpage-v2/).

**Reply by adamrher** [reply]
Is there a shared NCAR drive with more up to date stock images?

**Reply by cecilehannay** [reply]
@adamrher: I am not sure I understand the request.

**Reply by adamrher** [reply]
I just noticed the outdated NCAR icon on the website, and wondered if there is a shared drive with more up to date logos, or other misc NCAR images ...

**Reply by justin-richling** [reply]
I was looking for one myself, it would be great if we could find it

**Reply by dan800** [reply]
Ideally we'd be conforming to our "brand"
https://news.ucar.edu/brand

This is why I mentioned moving the website to Jekyll - this is where all CGD web content is being migrated to - templates for NCAR exist, I believe.

**Comment by swrneale** [comment]
A selection of logos is here https://drive.google.com/drive/folders/1fGb0CX-Hl_Hz2x8mhdmet98JjcVNd6y3 .
And this is a phenomenal document on corporate identity https://news.ucar.edu/sites/default/files/documents/related-links/2020-03/NCAR-UCAR_BrandStandards_031020-Spreads.pdf

**Reply by dan800** [reply]
I noticed there's no license for the ADF package listed on the GitHub repo. Do you know what license if falls under?

**Reply by cecilehannay** [reply]
@dan800: this is very good question. As far as I know, this has not been discussed yet. 
I will reach out to see how to proceed.

**Comment by swrneale** [comment]
GMD has code guidelines and possible license options at the bottom [here](https://www.geoscientific-model-development.net/policies/code_and_data_policy.html) which I think are useful.

**Reply by cecilehannay** [reply]
Thanks, Rich. 
We worked on the license for the amwg diags with the UCAR lawyer in 2014. They had very specific requirements and very specific wording.

**Reply by cecilehannay** [reply]
I talked to Gokhan and Gary.  They said we should use the CC BY 4.0 license for this purpose.

**Reply by brianpm** [reply]
CC BY 4.0 sounds like a good choice to me. This is very easy to add to a github repo; we can discuss it at the ADF meeting/hackathon.

**Reply by cecilehannay** [reply]
FYI: We just added the license.

**Comment by dan800** [comment]
I was just looking at the latest F case comparisons -
https://webext.cgd.ucar.edu/FWscHIST/f.cesm3_cam058_mom_e.FWscHIST.ne30_L58.024/atm/f.cesm3_cam058_mom_e.FWscHIST.ne30_L58.024_1979_1984_vs_f.cesm3_cam058_mom_e.FWscHIST.ne30_L58.023_1979_1984/html_img/plot_page_CLDLOW_ANN_Zonal_Mean.html
It would be nice to add a legend to indicate which color is control and which is test. Is the difference plot test - control? Also perhaps change the table columns to test and control rather than case and baseline?

**Comment by justin-richling** [comment]
Thanks for the input @dan800, I've just done an update that gives the plots a bit more info, some examples can be found at the most recent run you described: [https://webext.cgd.ucar.edu/FWscHIST/f.cesm3_cam058_mom_e.FWscHIST.ne30_L58.024/atm/f.cesm3_cam058_mom_e.FWscHIST.ne30_L58.024_1979_1990_vs_f.cesm3_cam058_mom_e.FWscHIST.ne30_L58.023_1979_1990/](https://webext.cgd.ucar.edu/FWscHIST/f.cesm3_cam058_mom_e.FWscHIST.ne30_L58.024/atm/f.cesm3_cam058_mom_e.FWscHIST.ne30_L58.024_1979_1990_vs_f.cesm3_cam058_mom_e.FWscHIST.ne30_L58.023_1979_1990/). 

There are some details still left, but I agree we should keep case/baseline/control all consistent in the tables and elsewhere, that is also something I'm currently looking at.

**Comment by dan800** [comment]
I just chatted with Justin and it sounds like he's one step ahead of me in implementing improvements (e.g., introducing a nickname for a run to simplify plot titles). One thing that we discussed (so mentioning it here) is to use the stats in the individual case tables to add a test for significance (e.g., Students t test) in the difference table, with color coding for items that pop out as being statistically different at a certain level.

**Comment by justin-richling** [comment]
Justin's ADF slides for CMA7 meeting on 12/02/2022

[2022-12-02 ADF Update.pdf](https://github.com/NCAR/ADF/files/10157886/2022-12-02.ADF.Update.pdf)

**Comment by adamrher** [comment]
FYI, E3SM's diag package paper is on gmdd: https://gmd.copernicus.org/preprints/gmd-2022-38/
